framework/src/ble_manager/ble_manager_state.c

```diff
@@ -542,14 +625,31 @@ ble_result_e blemgr_handle_request(blemgr_msg_s *msg)
 
 		int i;
 		ble_client_ctx *ctx = NULL;
+		ble_client_state_e priv_state = BLE_CLIENT_NONE;
 		for (i = 0; i < BLE_MAX_CONNECTION_COUNT; i++) {
-			if (g_client_table[i].state == BLE_CLIENT_CONNECTING && memcmp(g_client_table[i].info.addr.mac, data->conn_info.addr.mac, BLE_BD_ADDR_MAX_LEN) == 0) {
-				g_client_table[i].state = BLE_CLIENT_CONNECTED;
-				g_client_table[i].conn_handle = data->conn_handle;
-				g_client_table[i].is_bonded = data->is_bonded;
-				memcpy(&(g_client_table[i].info), &(data->conn_info), sizeof(ble_conn_info));
-				ctx = &g_client_table[i];
-				break;
+			if (memcmp(g_client_table[i].info.addr.mac, data->conn_info.addr.mac, BLE_BD_ADDR_MAX_LEN) == 0) {
+				if (g_client_table[i].state == BLE_CLIENT_CONNECTING || g_client_table[i].state == BLE_CLIENT_AUTOCONNECTING) {
+					priv_state = g_client_table[i].state;
+					g_client_table[i].state = BLE_CLIENT_CONNECTED;
+					g_client_table[i].conn_handle = data->conn_handle;
+					g_client_table[i].is_bonded = data->is_bonded;
+
+					memcpy(&(g_client_table[i].info.addr), &(data->conn_info.addr), sizeof(trble_addr));
+					g_client_table[i].info.conn_interval = data->conn_info.conn_interval;
+					g_client_table[i].info.slave_latency = data->conn_info.slave_latency;
+					g_client_table[i].info.mtu = data->conn_info.mtu;
+					ctx = &g_client_table[i];
+					break;
+				}
+			}
+		}
+
+		if (priv_state == BLE_CLIENT_AUTOCONNECTING) {
+			char buf[BLE_MQ_SIZE] = { 0, };
+			buf[0] = BLE_AUTOCON_EVT_CONNECTED;
+			int status = mq_send(ctx->mqfd, buf, BLE_MQ_SIZE, 0);
```

Comment:
Is it okay not to close the mq?

Reply:
The 'mqfd' will be closed when auto-connection thread is done.
(ble_manager_autoconnect.c Line: 185)